### PR TITLE
.github/workflows/update-terraform-providers.yml: init 

### PR DESCRIPTION
**Pull request by zowoq**
###### Motivation for this change

weekly update of terraform providers which can also be run manually

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Mic92** [comment]
First run: https://github.com/NixOS/nixpkgs/runs/4699277217?check_suite_focus=true

File: .github/workflows/update-terraform-providers.yml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+          popd
+      - name: create PR
+        uses: peter-evans/create-pull-request@v3
+        with:
+          body: |
+            Automatic update of terraform providers.
+
+            Created by [update-terraform-providers](https://github.com/NixOS/nixpkgs/blob/master/.github/workflows/update-terraform-providers.yml) action.
+
+            @ofborg build terraform-full
```
Comment (zowoq) [inline .github/workflows/update-terraform-providers.yml:35]: Seems this doesn't trigger `ofborg` for some reason.

**Comment by zimbatm** [comment]
Did you look into @ryantm's updater bot? Automation is key to scale but we also have to be careful not to introduce too many update methods.

**Comment by zowoq** [comment]
> Did you look into @ryantm's updater bot? 

Yes. We'd need to either teach it about the terraform registry or set it to run the script on a timer. That it isn't already being used to update the node and vim package sets makes me think that it's not simple to do.

> Automation is key to scale but we also have to be careful not to introduce too many update methods.

I think the bus factor of anything that is important to the NixOS org that is outside of this repo is a vastly bigger concern than any number of updaters that are in this repo.

**Pull request merged**